bin/propolis-utils/src/bin/cpuid-gen.rs

```diff
@@ -16,17 +16,17 @@ fn create_vm() -> anyhow::Result<VmmFd> {
     let ctl = VmmCtlFd::open()?;
     let _ = unsafe { ctl.ioctl(bhyve_api::VMM_CREATE_VM, &mut req) }?;
 
-    let vm = VmmFd::open(&name).map_err(|e| {
+    let vm = VmmFd::open(&name).inspect_err(|_e| {
         // Attempt to manually destroy the VM if we cannot open it
```

Comment:
this suggestion kinda makes my skin crawl. "inspect_err" sounds a lot like you should expect `println!()` or something boring to follow, but on the next like we are _destroying a resource_ because of the error. i'm not interested in adding `#![allow(clippy::manual_inspect)]` in an ad-hoc manner at crate roots though, so.. this what the suggested fix looks like.

Reply:
Optional, just noodling on the problem a bit: does it look any better to write it out by hand?

```rust
let vm = VmmFd::open(&name);
if vm.is_err() {
    let _ = ctl.vm_destroy(name.as_bytes());
}
let vm = vm?;
```

Or we could just write out the `?` match explicitly. I don't have especially strong feelings here--it does seem like `inspect` is not quite the right verb for what we're trying to do here, but I'm not too sure `map` was either, since this wasn't transforming the error at all. Up to you in any case.

Reply:
i'd given that one a try and realized i was having a hard time reading it after the fact, since the `?` and exit control flow is separated away from `.is_err()`. an explicit match would probably read fine though, i'll do that in a sec

Reply:
sorry for the drive-by comment, but might this be `or_else`?

    let vm = VmmFd::open(&name).or_else(|e| {
        let _ = ctl.vm_destroy(name.as_bytes());
        Err(e)
    })?;

To me, the `or_else` says "we're doing something if this fails."

Reply:
i'd have liked `or_else` a lot but it turns out that is also on the path to `inspect_err`: _that_ results in Clippy offering "using `Result.or_else(|x| Err(y))`, which is more succinctly expressed as `map_err(|x| y)`. it's right, but then with that transform Clippy points towards `inspect_err`. :upside_down_face: 

anyway, i've adjusted this to be an explicit match on both cases. seems fine.

Reply:
Aww Clippy, why ya gotta be like that? :-/